### PR TITLE
SAK-44532: Error when try adding a student on a blocked section/group

### PR DESCRIPTION
**Pull request by jesusmmp**
Please see description: https://jira.sakaiproject.org/browse/SAK-44532

Regression caused by: https://github.com/sakaiproject/sakai/commit/0b9d28b85062a7d6358c57bd0accedcdf9904a8b#diff-be390a19db9bd25a62515c17e11a06599ccc4a11744d6d98a9c9ad70315c8050R911

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java [hunk 1]
```diff
@@ -413,16 +413,16 @@ public void addMember(String userId, String roleId, boolean active, boolean prov
 	{
 		try {
 			insertMember(userId, roleId, active, provided);
-		} catch (IllegalStateException ise) {
-			log.warn(ise.getMessage());
+		} catch (IllegalStateException | AuthzRealmLockException isearle) {
```
Comment (bjones86) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java:416]: Let's not call this exception `isearle` :)

**Review by bjones86 — CHANGES_REQUESTED** [review]

File: kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java (hunk 1 above)
Comment (jesusmmp) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java]: Changed to iseorarle!

**Review by bjones86 — APPROVED** [review]

File: kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java [hunk 2]
```diff
@@ -413,16 +413,16 @@ public void addMember(String userId, String roleId, boolean active, boolean prov
 	{
 		try {
 			insertMember(userId, roleId, active, provided);
-		} catch (IllegalStateException ise) {
-			log.warn(ise.getMessage());
+		} catch (IllegalStateException | AuthzRealmLockException iseorarle) {
+			log.warn(iseorarle.getMessage());
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java:417]:
```suggestion
		} catch (AuthzRealmLockException arle) {
			log.warn(arle.toString());
```

File: kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java [hunk 3]
```diff
@@ -413,16 +413,16 @@ public void addMember(String userId, String roleId, boolean active, boolean prov
...
 		}
 	}
 
-	public void insertMember(String userId, String roleId, boolean active, boolean provided) throws IllegalStateException
+	public void insertMember(String userId, String roleId, boolean active, boolean provided) throws IllegalStateException, AuthzRealmLockException
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java:421]:
```suggestion
	public void insertMember(String userId, String roleId, boolean active, boolean provided) throws AuthzRealmLockException
```

**Review by ern — CHANGES_REQUESTED** [review]
Couple of updates

**Comment by ern** [comment]
Thanks for catching this I must have missed this method :(

File: kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java (hunk 2 above)
Comment (bjones86) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java:417]: What's the benefit of catching them separately?

**Comment by jesusmmp** [comment]
Suggested changes has been done!

**Review by ern — APPROVED** [review]

File: kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java [hunk 4]
```diff
@@ -415,14 +415,16 @@ public void addMember(String userId, String roleId, boolean active, boolean prov
 			insertMember(userId, roleId, active, provided);
 		} catch (IllegalStateException ise) {
 			log.warn(ise.getMessage());
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseGroup.java:417]:
```suggestion
```

**Pull request merged**